corgi/tasks/brew.py

```diff
@@ -701,3 +699,49 @@ def slow_refresh_brew_build_tags(build_id: int) -> None:
     for erratum_id in sorted(new_errata_tags):
         slow_load_errata.delay(erratum_id)
     logger.info(f"Finished refreshing Brew build tags for {build_id}")
+
+
+@app.task(base=Singleton, autoretry_for=RETRYABLE_ERRORS, retry_kwargs=RETRY_KWARGS)
+def slow_delete_brew_build(build_id: int, build_state: int) -> int:
+    """Delete a Brew build (and its relations) in Corgi when it's deleted in Brew"""
+    # Shipped builds (which have a Brew tag for any product) are never deleted
+    # Only unshipped builds not part of any product are deleted
+    # For example, builds that failed QE will not become part of a product
+    # We don't need this data for any of our use cases
+    # Keeping it causes other issues when reloading old builds, scanning for licenses, etc.
+
+    logger.info(f"Deleting Brew build {build_id} with state {build_state}")
+    if build_state != koji.BUILD_STATES["DELETED"]:
+        raise ValueError(
+            f"Invalid state for build {build_id}: "
+            f"expected {koji.BUILD_STATES['DELETED']}, received {build_state}"
+        )
+
+    # When we call .delete() on a queryset, Django returns a tuple with 2 elements
+    # The first element is the number of models we deleted directly / the queryset length
+    # The second element is a dict. The keys are this model's name, plus all the related model names
+    # which have a ForeignKey to the deleted model that defines on_delete=models.CASCADE
+    # The values are the number of model instances we deleted either directly or due to a ForeignKey
+    with transaction.atomic():
+        deleted_info: tuple[int, dict[str, int]] = ProductComponentRelation.objects.filter(
+            build_id=build_id, build_type=SoftwareBuild.Type.BREW
+        ).delete()
```

Comment:
Just curious: why don't we have a cascade delete for ProductComponentRelation that would handle this? Is there a case where the relation should remain in place even if it links to a non-existing build?